lib/errors.js

```diff
@@ -1,39 +1,73 @@
+'use strict';
```

Comment:
the only place ```'use strict';``` doesn't work currently is in the schemautils (lib/schema.js), and that's because of a recursive this reference iirc. 

I'm not sure I agree with you about ```'use strict';```, how does it take more time than it saves? Frankly it was basically as easy as adding it to the top of each file, and have that locked in and enforced by jshint.. took no time at all, and you get these benefits from it: 

"Strict mode makes several changes to normal JavaScript semantics. First, strict mode eliminates some JavaScript silent errors by changing them to throw errors. Second, strict mode fixes mistakes that make it difficult for JavaScript engines to perform optimizations: strict mode code can sometimes be made to run faster than identical code that's not strict mode. Third, strict mode prohibits some syntax likely to be defined in future versions of ECMAScript." (from [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode))